controllers/factory/alertmanager/config.go

```diff
@@ -798,13 +796,18 @@ func (cb *configBuilder) buildPagerDuty(pd operatorv1beta1.PagerDutyConfig) erro
 
 func (cb *configBuilder) buildEmail(email operatorv1beta1.EmailConfig) error {
 	var temp yaml.MapSlice
-	if email.TLSConfig != nil {
+	if email.RequireTLS != nil {
+		temp = append(temp, yaml.MapItem{Key: "require_tls", Value: *email.RequireTLS})
+	}
+	// not set tls_config if require_tls is false
+	if email.RequireTLS == nil || *email.RequireTLS {
```

Comment:
It looks that this part could panic in cases:
 *email.RequireTLS == true and email.TLSConfig == nil.

 Probably, it's better to add sanity check
```go
if email.RequireTLS == nil ||  *email.RequireTLS {
   if email.TLSConfig == nil {
             return fmt.Errorf("incorrect email configuration, tls is required, but no config provided at spec")
   }
}
```